src/cli.rs

```diff
@@ -213,11 +213,12 @@ pub struct RootOpts {
     #[arg(long, env = "VECTOR_ALLOW_EMPTY_CONFIG", default_value = "false")]
     pub allow_empty_config: bool,
 
-    /// Turn on strict mode for environment variable interpolation. When set, interpolation of a
-    /// missing environment variable in configuration files will cause an error instead of a
-    /// warning, which will result in a failure to load any such configuration file. This defaults
-    /// to false, but that default is deprecated and will be changed to strict in future versions.
-    #[arg(long, env = "VECTOR_STRICT_ENV_VARS", default_value = "false")]
+    /// Turn on strict mode for environment variable interpolation. When set, interpolation of
+    /// a missing environment variable in configuration files will cause an error instead of
+    /// a warning, which will result in a failure to load any such configuration file. This option
+    /// is deprecated and will be removed in a future version to remove the ability to downgrade
+    /// missing environment variables to warnings.
+    #[arg(long, env = "VECTOR_STRICT_ENV_VARS", default_value = "true")]
     pub strict_env_vars: bool,
```

Comment:
Is there any way to turn _off_ strict mode with this change?

Reply:
I confirmed this with a short test program — with this change, there is no way to disable strict mode from the command line, only using the environment variable. I'm going to pull it from the merge queue until we confirm that we want to drop this ability from the command line.

Reply:
Resolved by [807194c](https://github.com/vectordotdev/vector/pull/20062/commits/807194c7a5e84600eff20302005481469a58414e)